Sources/CloudKitSyncMonitor/SyncMonitor.swift

```diff
@@ -494,6 +494,13 @@ public class SyncMonitor: ObservableObject {
     /// When SyncMonitor is listening to notifications (which it does unless you tell it not to when initializing), this method is called each time CKContainer
     /// fires off a `.CKAccountChanged` notification.
     private func updateiCloudAccountStatus() {
+        #if DEBUG
+        let isPreview = ProcessInfo.processInfo.environment["XCODE_RUNNING_FOR_PREVIEWS"] == "1"
+        guard !isPreview else {
+            return
+        }
```

Comment:
Hi @longvudai ,

Thanks for the bug fix.

Can you rewrite the "guard" line to something more direct? I had to re-read the code many times to figure out what it does due to the multiple negatives (i.e., I read it as "let isPreview equal xcode running for previews is true; make sure we're not in a preview, otherwise, return").

I think this would do it (but make sure you test it, don't just copy/paste, because I don't have a Mac handy with which to test this at the moment :) ):

```swift
#if DEBUG
// Prevent a crash in SwiftUI previews
let isPreview = ProcessInfo.processInfo.environment["XCODE_RUNNING_FOR_PREVIEWS"] == "1"
if isPreview { return }
#endif
```

I realize that "guard" is usually preferable when aborting a function like this, but I think in this case we either need "if" or a more direct `guard`.

Thanks!

Grant

Reply:
@ggruen  I managed to resolve it and successfully put it to the test. 
![Screenshot 2024-08-05 at 10 55 13](https://github.com/user-attachments/assets/2476965c-d35c-4649-97ac-4b27940a8ea1)